include/exadg/operators/inverse_mass_operator.h

```diff
@@ -60,14 +62,30 @@ class InverseMassOperator
     this->matrix_free = &matrix_free_in;
     dof_index         = dof_index_in;
     quad_index        = quad_index_in;
+
+    explicit_matrix_free_inverse_mass_available = matrix_free->get_dof_handler(dof_index)
+                                                    .get_triangulation()
+                                                    .all_reference_cells_are_hyper_cube();
+
+    if(not(explicit_matrix_free_inverse_mass_available))
+    {
+      initialize_inverse_mass_operator_with_block_jacobi();
+    }
   }
 
   void
   apply(VectorType & dst, VectorType const & src) const
   {
     dst.zero_out_ghost_values();
 
-    matrix_free->cell_loop(&This::cell_loop, this, dst, src);
+    if(not(explicit_matrix_free_inverse_mass_available))
+    {
+      mass_preconditioner->vmult(dst, src);
+    }
+    else
+    {
+      matrix_free->cell_loop(&This::cell_loop, this, dst, src);
+    }
```

Comment:
You can get rid of the `not` by changing the order of the two branches. This makes the code much easier to read.